apps/experiments/views/experiment.py

```diff
@@ -1255,6 +1324,76 @@ def experiment_session_messages_view(request, team_slug: str, experiment_id: uui
     )
 
 
+@experiment_session_view()
+@verify_session_access_cookie
+def translate_messages_view(request, team_slug: str, experiment_id: uuid.UUID, session_id: str):
+    from apps.analysis.translation import translate_messages_with_llm
+
+    session = request.experiment_session
+    language = request.POST.get("language")
+    provider_model = request.POST.get("provider_model", "")
+    valid_languages = [choice[0] for choice in LANGUAGE_CHOICES if choice[0]]
+
+    if not language or language not in valid_languages:
+        messages.error(request, "No language selected for translation.")
+        return redirect_to_messages_view(request, session)
+    if not provider_model:
+        messages.error(request, "No LLM provider model selected.")
+        return redirect_to_messages_view(request, session)
+    try:
+        provider_id, model_id = provider_model.split(":", 1)
+        llm_provider = LlmProvider.objects.get(id=provider_id, team=request.team)
+        llm_provider_model = LlmProviderModel.objects.get(id=model_id)
+
+        messages_to_translate = ChatMessage.objects.filter(chat=session.chat).exclude(
+            **{f"translations__{language}__isnull": False}
+        )
+
+        if not messages_to_translate.exists():
+            messages.info(request, "All messages already have translations for this language.")
+            return redirect_to_messages_view(request, session)
+
+        translate_messages_with_llm(
+            messages=list(messages_to_translate),
+            target_language=language,
+            llm_provider=llm_provider,
+            llm_provider_model=llm_provider_model,
+        )
+    except ValueError:
+        messages.error(request, "Invalid provider model format.")
+        return redirect_to_messages_view(request, session)
+    except (LlmProvider.DoesNotExist, LlmProviderModel.DoesNotExist):
```

Comment:
Where possible it's good practice to scope the exception handling to only the code where that exception can be raised. /in this case you could wrap just the code that fetches the provider and model.